### PR TITLE
[gfx90a] Direct conv3x3 wrw support

### PR DESCRIPTION
**Pull request by Slimakanzer**
This PR adds Direct conv3x3wrw support for gfx90a. Partially resolves issue ROCmSoftwarePlatform/MIOpen-internal#2.

## Local verification
- gfx90a, rocm 4.3.1

<details><summary>Test cases</summary>

type | -W | -H | -c | -n | -k | -x | -y | -p | -q | -u | -v | -F | -g
--   | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
fp16/fp32 | 7 | 7 | 160 | 32 | 224 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 1
fp16/fp32 | 7 | 7 | 160 | 32 | 224 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 4
fp16/fp32 | 7 | 7 | 160 | 32 | 320 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 1
fp16/fp32 | 7 | 7 | 160 | 32 | 320 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 4
fp16/fp32 | 7 | 7 | 160 | 128 | 320 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 1
fp16/fp32 | 7 | 7 | 160 | 128 | 320 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 4
fp16/fp32 | 7 | 7 | 192 | 32 | 224 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 1
fp16/fp32 | 7 | 7 | 192 | 32 | 224 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 4
fp16/fp32 | 7 | 7 | 192 | 32 | 320 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 1
fp16/fp32 | 7 | 7 | 192 | 32 | 320 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 4
fp16/fp32 | 13 | 13 | 192 | 50 | 384 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 1
fp16/fp32 | 13 | 13 | 192 | 50 | 384 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 4
fp16/fp32 | 13 | 13 | 192 | 64 | 384 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 1
fp16/fp32 | 13 | 13 | 192 | 64 | 384 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 4
fp16/fp32 | 14 | 14 | 96 | 32 | 128 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 1
fp16/fp32 | 14 | 14 | 96 | 32 | 128 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 4
fp16/fp32 | 14 | 14 | 96 | 32 | 208 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 1
fp16/fp32 | 14 | 14 | 96 | 32 | 208 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 4
fp16/fp32 | 14 | 14 | 160 | 32 | 192 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 1
fp16/fp32 | 14 | 14 | 160 | 32 | 192 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 4
fp16/fp32 | 14 | 14 | 160 | 32 | 320 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 1
fp16/fp32 | 14 | 14 | 160 | 32 | 320 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 4
fp16/fp32 | 27 | 27 | 128 | 1 | 128 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 1
fp16/fp32 | 27 | 27 | 128 | 1 | 128 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 4
fp16/fp32 | 27 | 27 | 128 | 8 | 128 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 1
fp16/fp32 | 27 | 27 | 128 | 8 | 128 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 4
fp16/fp32 | 28 | 28 | 256 | 8 | 512 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 1
fp16/fp32 | 28 | 28 | 256 | 8 | 512 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 4
fp16/fp32 | 28 | 28 | 256 | 16 | 512 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 1
fp16/fp32 | 28 | 28 | 256 | 16 | 512 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 4
fp16/fp32 | 56 | 56 | 64 | 8 | 64 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 1
fp16/fp32 | 56 | 56 | 64 | 8 | 64 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 4
fp16/fp32 | 56 | 56 | 64 | 32 | 192 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 1
fp16/fp32 | 56 | 56 | 64 | 32 | 192 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 4
fp16/fp32 | 56 | 56 | 64 | 128 | 192 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 1
fp16/fp32 | 56 | 56 | 64 | 128 | 192 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 4
fp16/fp32 | 56 | 56 | 128 | 1 | 256 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 1
fp16/fp32 | 56 | 56 | 128 | 1 | 256 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 4
fp16/fp32 | 56 | 56 | 128 | 2 | 256 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 1
fp16/fp32 | 56 | 56 | 128 | 2 | 256 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 4
fp16/fp32 | 56 | 56 | 128 | 8 | 256 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 1
fp16/fp32 | 56 | 56 | 128 | 8 | 256 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 4
fp16/fp32 | 120 | 12 | 32 | 1 | 64 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 1
fp16/fp32 | 120 | 12 | 32 | 1 | 64 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 4
fp16/fp32 | 350 | 80 | 64 | 1 | 64 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 1
fp16/fp32 | 350 | 80 | 64 | 1 | 64 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 4
fp16/fp32 | 350 | 80 | 64 | 2 | 64 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 1
fp16/fp32 | 350 | 80 | 64 | 2 | 64 | 3 | 3 | 1 | 1 | 1 | 1 | wrw | 4

</details>

### CODE REVIEW

**Comment by atamazov** [comment]
@JehandadKhan This enables tunable solver for gfx90a, so perf tuning is required for that solver + find-db update for gfx90a.

File: src/kernels/conv3x3wrw.s [hunk 1]
```diff
@@ -77,6 +77,21 @@ default reverse_inout, 0
...
+// gfx90a requires 64bit aligned vgpr tuples
+// Tuples are used only in buffer_load_dwordx/buffer_store_dwordx instructions
+//
+// To meet this requirement, the following approach is used ('buffer_load_dwordx4 v[x:y]' as an example):
+//    if 'x' 64bit aligned:
+//       buffer_load_dwordx4 v[x:y], ...
+//    if 'x' not 64bit aligned:
+//       buffer_load_dword   v[x], ...
+//       buffer_load_dwordx3 v[x+1:y], ...
+.if (.amdgcn.gfx_generation_number == 9 && .amdgcn.gfx_generation_stepping == 10)
```
Comment (atamazov) [inline src/kernels/conv3x3wrw.s:89]: [Recommendation] This enables alignment for `9*a`, shall we better enable it for `90a`?

**Review by atamazov — APPROVED** [review]
Good to go.

File: src/solver/conv_asm_dir_BwdWrW3x3.cpp [hunk 2]
```diff
@@ -38,7 +38,6 @@
 #include <miopen/generic_search.hpp>
 
 #define WORKAROUND_ISSUE_532 1 // ConvAsmBwdWrW3x3 has precision issues with some PerformanceConfigs
-#define WORKAROUND_ISSUE_1146 1 // check asm solver applicability for gfx90a
```
Comment (junliume) [inline src/solver/conv_asm_dir_BwdWrW3x3.cpp:41]: Thanks! Put 1 in the can and 8 more to go :-)

**Review by junliume — APPROVED** [review]

File: src/solver/conv_asm_dir_BwdWrW3x3.cpp (hunk 2 above)
Comment (atamazov) [reply src/solver/conv_asm_dir_BwdWrW3x3.cpp:41]: @junliume This PR shouldn't be necessary for that SWDEV.

**Pull request merged**